qiskit_experiments/calibration/calibration_definitions.py

```diff
@@ -0,0 +1,382 @@
+        """
+        Return the schedules in self in a data frame to help
+        users manage their schedules.
+
+        Returns:
+            data: A pandas data frame with all the schedules in it.
+        """
+        data = []
+        for name, schedule in self._schedules.items():
+            data.append({'name': name,
+                         'schedule': schedule,
+                         'parameters': schedule.parameters})
+
+        return pd.DataFrame(data)
+
+    def parameters(self, names: Optional[List[str]] = None,
+                   chs: Optional[List[PulseChannel]] = None) -> pd.DataFrame:
+        """
+        Returns the parameters as a pandas data frame.
+        This function is here to help users manage their parameters.
+
+        Args:
+            names: The parameter names that should be included in the returned
+                table. If None is given then all names are included.
+            chs: The channels that should be included in the returned table.
+                If None is given then all channels are returned.
+
+        Returns:
+            data: A data frame of parameter values.
+        """
+
+        data = []
+
+        if names is None:
+            names = self._params.keys()
+
+        for name in names:
+            params_name = self._params.get(name, {})
+
+            if chs is None:
+                chs = params_name.keys()
+
+            for ch in chs:
+                for value in params_name.get(ch, {}):
+                    value_dict = dataclasses.asdict(value)
+                    value_dict['channel'] = ch.name
+                    value_dict['parameter'] = name
+
+                    data.append(value_dict)
+
+        return pd.DataFrame(data)
+
+    def add_schedules(self, schedules: Union[Schedule, List[Schedule]]):
+        """
+        Add a schedule and register the parameters.
+
+        Args:
+            schedules: The schedule to add.
+
+        Raises:
+            CalibrationError: If the parameterized channel index is not formatted
+                following index1.index2...
+        """
+        if isinstance(schedules, Schedule):
+            schedules = [schedules]
+
+        for schedule in schedules:
+
+            # check that channels, if parameterized, have the proper name format.
+            #pylint: disable = raise-missing-from
+            for ch in schedule.channels:
+                if isinstance(ch.index, Parameter):
+                    try:
+                        [int(_) for _ in ch.index.name.split('.')]
+                    except ValueError:
+                        raise CalibrationError('Parameterized channel must have a name '
+                                               'formatted following index1.index2...')
+
+            self._schedules[schedule.name] = schedule
+
+            for param in schedule.parameters:
+                if param.name not in self._params:
+                    self._params[param.name] = {}
```

Comment:
Users need to manage parameter names for every schedule so that not overlapping with existing name, when constructing schedule? I think this is very annoying.

Reply:
I think this is acceptable as in most cases we only need a very limited number of schedules. If you have a better alternative to suggest please do so.